### PR TITLE
optimize batch normalization and relu layer

### PR DESCRIPTION
**Pull request by hjchen2**
Merged the BatchNorm and Scale layer.
Optimize most elapsed time of batch normalization by re-writing the forward and backward process using OpenMP rather than BLAS to avoid additional multiply operation.
The elapsed time of ResNet-50 with batch size set by 128 fells from 32523.77ms to 16982.93ms on Intel(R) Xeon(R) CPU E5-2450 0 @ 2.10GHz.

### CODE REVIEW

**Comment by jczaja** [comment]
Thanks very much we will have it reviewed shortly!

**Comment by tpatejko** [comment]
Thank you for your effort.

We reviewed the pull request and we are not going to merge it right now. 

The code changes functionality of Intel Caffe:
- If I understand correctly, the pull request implements shift and scale in batch normalization. Original implementation separates these two operations;
- The pull request removes code guarded by variable use_global_stats. It might influence the performance of trained network.
